lib/modules/deployment/index.js

```diff
@@ -81,25 +81,20 @@ class DeployManager {
               contractDeploys[className].push(deploy);
             });
 
-            try {
-              async.auto(contractDeploys, 1, function(_err, _results) {
-                if (errors.length) {
-                  _err = __("Error deploying contracts. Please fix errors to continue.");
-                  self.logger.error(_err);
-                  self.events.emit("outputError", __("Error deploying contracts, please check console"));
-                  return done(_err);
-                }
-                if (contracts.length === 0) {
-                  self.logger.info(__("no contracts found"));
-                  return done();
-                }
-                self.logger.info(__("finished deploying contracts"));
-                done(err);
-              });
-            } catch (e) {
-              self.logger.error(e.message || e);
-              done(__('Error deploying'));
-            }
+            async.auto(contractDeploys, 1, function(_err, _results) {
```

Comment:
I feel we don't need auto with 1 :)
But might be for later

Reply:
I honestly am not aware of what async.auto does and didn't look further into the sense of using this API in the first place.

If you think this should be changed, I'm happy to do so.

Ideally we get rid off `async()` and rely on async/await or observables in the long run..

Reply:
The `1` is the concurrency.
So that will always be a normal `forEach` AFAIK
@jrainville can probably help here

Reply:
Hello. Auto is really nice in the wait that you give it an array/object of stuff to iterate over with dependencies and it will automatically determine the order to loop on based on the dependencies.
So we basically used that to parallely deploy our contracts.
Then horror struck: we got nonce problems, and we had to kill the parallel deploy. That's why we put `1` there, so that it deploys normally. 
We need to find a way to fix it. To be fair, the problem happened when I was in vacation so I never had the chance to test with it.